modules/user_accounts/php/edit_user.class.inc

```diff
@@ -973,7 +973,21 @@ class Edit_User extends \NDB_Form
 
                 if (strlen($effectiveUID) > 255) {
                     $errors['UserID_Group']
-                        = 'The user name must not exsceed 255 characters';
+                        = 'The user name must not exceed 255 characters';
+                }
+
+                // Check that user name does not contain a whitespace character
+                if (preg_match('/\s/', $effectiveUID)) {
+                    // Note: email adresses can contain comments which themselves can
+                    // contain spaces
+                    if ($values['NA_UserID'] == 'on') {
+                        $errors['UserID_Group']
+                            = 'You cannot have the user name match an email adress'
```

Comment:
```suggestion
                            = 'You cannot have the user name match an email address'
```

Small typo.

Also I'm a little bit confused by this part of the code. What is this checking for?

Reply:
@johnsaigle I am trying to display an error message that is coherent with the way that the user name is set (i.e whether it was typed in the user name text field or it the 'Make user name match email address' checkbox was used).